sys/server/src/TxsServer.hs

```diff
@@ -81,21 +81,22 @@ main = withSocketsDo $ do
       hPutStrLn stderr "Errors found while loading the configuration"
       hPrint stderr xs
     Right config -> do
-        (portNr, sock) <- txsListenOn $ (clPortNumber . SC.cmdLineCfg) uConfig
-        (hs, host, _) <- accept sock
-        hSetBuffering hs LineBuffering
-        hSetEncoding hs latin1
-        hPutStrLn stderr "\nTXSSERVER >>  Starting  ..... \n"
-        let initS = IOS.envsNone
-                { IOS.host   = host
-                , IOS.portNr = portNr
-                , IOS.servhs = hs
-                }
-            coreConfig = config
-        TxsCore.runTxsCore coreConfig cmdsIntpr initS
-        threadDelay 1000000    -- 1 sec delay on closing
-        sClose sock
-        hPutStrLn stderr "\nTXSSERVER >>  Closing  ..... \n"
+      (portNr, sock) <- txsListenOn $ (clPortNumber . SC.cmdLineCfg) uConfig
+      (hs, host, _) <- accept sock
+      hSetBuffering hs LineBuffering
+      hSetEncoding hs latin1
+      hPutStrLn stderr "\nTXSSERVER >>  Starting  ..... \n"
+      let initS = IOS.envsNone
+              { IOS.host   = host
+              , IOS.portNr = portNr
+              , IOS.servhs = hs
+              , IOS.params = TxsCore.updateParams (IOS.params IOS.envsNone) $ SC.configuredParameters config
```

Comment:
To me it looks strange that we have to call the `updateParams` function twice. We're calling `runTxsCore` below, which will do this again. I know you @keremispirli explained me why, but maybe it a good idea to:

    - include a comment on why this is done.
    - if there's no good reason for updating the parameters twice submit an issue to fix this.

Reply:
Added comments.
Parameters come from 3 different places, 2 are merged in TxsCore and the other is initial state in TxsServer, which later merges to others in `runStateT`. We can discuss the reason for keeping them separated with Pierre.